src/moi/constraints.jl

```diff
@@ -0,0 +1,52 @@
+set_bounds(info::Union{VariableInfo, ConstraintInfo}, bounds::Bounds) = set_bounds(info, MOI.Interval(bounds))
+
+function set_bounds(info::Union{VariableInfo, ConstraintInfo}, interval::MOI.Interval)
+    l, u = interval.lower, interval.upper
+    l == -Inf || (info.lower_bound = l)
```

Comment:
You mean explicitly setting `-Inf` / `Inf` as a lower / upper bound if it's in the interval? I guess I wasn't quite sure whether BARON would even allow using infinite values as bounds.

Reply:
I don't think you can explicitly set a bound to infinity in a bar file, but you can omit variable bounds and have BARON attempt to infer some itself. I'm not entirely sure if it's possible for the ``info`` bounds to disagree with the ``interval`` bounds in your implementation, but if so it's preferable to respect the change rather than silently ignore it.

Reply:
Note that in the current implementation, the way we handle e.g. `GreaterThan` is to convert it to an `Interval` with infinite upper bound first. So it may very well be that the user never explicitly provided the infinite upper bound. And note that since we now implement the allocate-load API, this is never used to modify an existing constraint. But I now do recognize that this has a bug. If there are two `SingleVariable`-in-`Bounds` constraints, then the bounds could actually be widened by adding the second constraint. I don't *think* JuMP ever creates two `SingleVariable`-in-`Bounds` constraints for one variable, but I also don't think MOI explicitly forbids it. Perhaps we should check to see if variable bounds have already been set in the `VariableInfo` and just convert to `ScalarAffineFunction`-in-`Bounds` otherwise. That or just error.

Reply:
I'll leave it to your judgement what to do here, just wanted to point out the potential for unexpected behavior. IMO an error is perfectly fine.